/*
 * Copyright (C) 2025 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package perfetto.protovm.test.protos;

message Element {
  optional int32 id = 1;
  optional int32 value = 2;
  optional fixed32 value_fixed32 = 3;
  optional fixed64 value_fixed64 = 4;
}

message TraceEntry {
  repeated Element elements = 1;
}

message ElementAndIndex {
  optional Element element = 1;
  optional int32 index = 2;
}

message Patch {
  repeated int32 elements_to_delete_by_id = 1;
  repeated Element elements_to_merge = 2;
  repeated Element elements_to_set = 3;
  repeated int32 elements_to_delete_by_index = 4;
  repeated ElementAndIndex elements_to_insert_by_index = 5;
}
